app/src/main/java/com/cmput301w23t40/capturetheqr/ScoreboardActivity.java

```diff
@@ -15,21 +16,38 @@
 import android.widget.TextView;
 import android.widget.Toast;
 
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+import android.Manifest;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 
 /**
  * This class defines the UI page for the QR Code Library
  */
-public class ScoreboardActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {
+public class ScoreboardActivity extends AppCompatActivity {
 
     private ListView listView;
     private ArrayAdapter<Player> playerAdapter;
     private Player my_player;
-    private String metric = "Highest QR score";
+    private String deviceID;
+    private PlayerLocation locationHelper;
 
+    private ActivityResultLauncher<String[]> locationPermissionLauncher = registerForActivityResult(
+            new ActivityResultContracts.RequestMultiplePermissions(), result -> {
+        /* result is a map with the permission name as the key and the value being a
+        boolean indicating whether the permission was granted */
+                // User granted location permissions
+                if (result.get("android.permission.ACCESS_COARSE_LOCATION") &&
+                        result.get("android.permission.ACCESS_COARSE_LOCATION")) {
```

Comment:
We don't need two checks for coarse location permissions. That was an typo in my original code that I had sent you before, but it is now fixed in AddQRActivity which changes the second one checks `ACCESS_FINE_LOCATION`. 

Though, for the purposes of checking a region we probably don't need fine location permissions (accurate to a 10-50 meter distance) and could probably just use coarse location (accurate to within a 3 km^2  area) https://developer.android.com/training/location/permissions#accuracy

Reply:
done